newIDE/app/src/EventsFunctionsExtensionEditor/OptionsEditorDialog/UsersAutocomplete.js

```diff
@@ -0,0 +1,149 @@
+        setLoading(false);
+      } catch (err) {
+        setLoading(false);
+        setAuthorsError('Error while loading users');
+        console.log(err);
+      }
+    },
+    [authorInput]
+  );
+
+  // call every time the authorInput changes
+  React.useEffect(
+    () => {
+      searchUserPublicProfiles();
+    },
+    [searchUserPublicProfiles, authorInput]
+  );
+
+  // do only once, the first time it is loaded
+  const getUserPublicProfilesForAutocomplete = React.useCallback(
+    async () => {
+      setAuthorsError(null);
+      const authorIds = props.eventsFunctionsExtension
+        .getAuthorIds()
+        .toJSArray();
+      if (!authorIds.length) {
+        setAuthors([]);
+        return;
+      }
+      try {
+        setLoading(true);
+        const userPublicProfilesByIds = await getUserPublicProfilesByIds(
+          authorIds
+        );
+        setAuthors(
+          Object.keys(userPublicProfilesByIds).map(userId => {
+            const userPublicProfile: UserPublicProfile =
+              userPublicProfilesByIds[userId];
+            if (userPublicProfile.username && userPublicProfile.id) {
+              return {
+                text: userPublicProfile.username,
+                value: userPublicProfile.id,
+              };
+            }
+
+            return null;
+          })
+        );
+        setLoading(false);
+      } catch (err) {
+        setLoading(false);
+        setAuthorsError('Error while loading the authors');
+        console.log(err);
+      }
+    },
+    [props.eventsFunctionsExtension]
+  );
+
+  // do only once
+  React.useEffect(
+    () => {
+      getUserPublicProfilesForAutocomplete();
+    },
+    [getUserPublicProfilesForAutocomplete]
+  );
+
+  return (
+    <SemiControlledMultiAutoComplete
+      hintText={t`Start typing a username`}
+      floatingLabelText={t`Authors`}
+      helperText={t`Select all the extensions contributors usernames. They will be displayed in the order selected.`}
+      value={authors}
+      onChange={(event, values) => {
+        if (values) {
+          // change authors in state
+          setAuthors(values);
+          // change authors in project
+          const authors = props.eventsFunctionsExtension.getAuthorIds();
+          authors.clear();
+          values.forEach(({ text, value }) => {
+            authors.push_back(value);
+          });
+          forceUpdate();
+        }
+      }}
+      inputValue={authorInput}
+      onInputChange={(event, value) => {
+        setAuthorInput(value);
+      }}
+      dataSource={completionUserPublicProfiles.map(
+        (userPublicProfile: UserPublicProfile) => {
+          if (userPublicProfile.username && userPublicProfile.id) {
+            return {
+              text: userPublicProfile.username,
+              value: userPublicProfile.id,
+            };
+          }
+
+          return null;
+        }
+      )}
+      loading={loading}
+      fullWidth
+      error={authorsError}
```

Comment:
Here you could call a helper function `getErrorMessage(error: ?Error)` (defined outside of the component, no need to keep it inside), that can look at if there is an error, and if yes return an error message.
This make it easy to improve it in the future to analyze the error (is this an axios response? => nice I can look at the status code and give a good error. is this a status code that I don't know or undefined => generic error message)

Reply:
in this case, the authorsError is already a string to be displayed as is, but yes I'll extract it so that we know we can modify it easily